src/main/java/de/medizininformatik_initiative/process/data_transfer/service/ReadData.java

```diff
@@ -70,14 +99,25 @@ protected void doExecute(DelegateExecution execution, Variables variables)
 		{
 			logger.warn(
 					"Could not read data-set on FHIR server with baseUrl '{}' for DMS '{}' and project-identifier '{}' referenced in Task with id '{}' - {}",
-					fhirClient.getFhirBaseUrl(), dmsIdentifier, projectIdentifier, task.getId(),
+					fhirClientFactory.getFhirBaseUrl(), dmsIdentifier, projectIdentifier, task.getId(),
 					exception.getMessage());
 
 			String error = "Read data-set failed - " + exception.getMessage();
 			throw new RuntimeException(error, exception);
 		}
 	}
 
+	private Resource getResource(DataResource attachment) throws IOException
+	{
+		// TODO handle stream directly and do not parse to resource
+		if (attachment.hasInputStream())
+			return new Binary().setData(attachment.stream.readAllBytes()).setContentType(attachment.mimetype);
```

Comment:
If we create a Binary resource here and later store the resource in a process variable, the resource will be serialized between this and the next BPMN task. This would move the size constraint problem into the BPMN and thus may not solve #34.
Storing byte array and mime type separately might be a workaround to at least get rid of the base64 encoding step. The correct solution would store the URL to binary here and later pass the "read" InputStream directly to the FHIR client storing the resource.

Reply:
Implemented in 94db4a5 and 72adeee